File: testbin.cpp
#include <iostream>
#include <fstream>
using namespace std;

struct Vertex {
    float x;
    float y;
    float z;
};

struct Normal {
    float x;
    float y;
    float z;
};

struct Face {
    int a;
    int b;
    int c;
};

struct Material {
    char r;
    char g;
    char b;
};

struct Config {
    float cpx,cpy,cpz;
    float crx,cry,crz;
    int vc,nc,fc,mc;
};

int main() {
    ifstream in("prism.bin",ios::out|ios::binary);
    Config config;
    in.read((char *)&config,sizeof(Config));
    cout << config.cpx << "\t" << config.vc << endl;
    in.close();
    return 0;
}

File: testbin.go
package main

import "log"
import "os"
import "bytes"
import "encoding/binary"

type config struct {
	cpx float32
	cpy float32
	cpz float32
	crx float32
	cry float32
	crz float32
	vc int32
	nc int32
	fc int32
	mc int32
}

func main() {
	file,err := os.Create("prism.bin")
	defer file.Close()
	if err!=nil {log.Fatal(err)}
	c := &config{17,0,0,0,0,0,98234,0,0,0}
	var buf bytes.Buffer
	binary.Write(&buf,binary.LittleEndian,c)
	_,err = file.Write(buf.Bytes())
	if err!=nil {log.Fatal(err)}
}